tests/integration/rerender-test.js

```diff
@@ -0,0 +1,147 @@
+import { setComponentTemplate } from '@ember/component';
+import templateOnly from '@ember/component/template-only';
+import { precompileTemplate } from '@ember/template-compilation';
+import {
+  find,
+  getSettledState,
+  render,
+  rerender,
+  setupContext,
+  setupRenderingContext,
+  teardownContext,
+  waitFor,
+} from '@ember/test-helpers';
+import { buildWaiter } from '@ember/test-waiters';
+import GlimmerComponent from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { hbs } from 'ember-cli-htmlbars';
+import { module, test } from 'qunit';
+
+module('rerender real-world', function (hooks) {
+  hooks.beforeEach(async function () {
+    await setupContext(this);
+    await setupRenderingContext(this);
+  });
+  hooks.afterEach(async function () {
+    await teardownContext(this);
+  });
+
+  test('rerender and settled describe different scenarios', async function (assert) {
+    let waiter = buildWaiter('test:wat');
+    let token;
+    let deferredResolve;
+    let promise;
+
+    class MyComponent extends GlimmerComponent {
+      @tracked value = 'not started';
+
+      constructor() {
+        super(...arguments);
+
+        promise = new Promise((resolve) => {
+          deferredResolve = resolve;
+          this.value = 'loading';
+          token = waiter.beginAsync();
+        });
+
+        promise.then(() => {
+          waiter.endAsync(token);
+          this.value = 'done';
+        });
+      }
+    }
+
+    setComponentTemplate(
+      hbs`<div data-test-content>The profile for {{@name}} is {{this.value}}</div>`,
+      MyComponent
+    );
+
+    const somePerson = new (class {
+      @tracked name = 'Zoey';
+    })();
+
+    const template = precompileTemplate(
+      '<MyComponent @name={{somePerson.name}} />',
+      {
+        scope() {
+          return {
+            somePerson,
+            MyComponent,
+          };
+        },
+      }
+    );
+
+    const component = setComponentTemplate(template, templateOnly());
+
+    const renderPromise = render(component);
+
+    // No actual render
+    assert.equal(this.element.textContent, '', 'Nothing on the screen yet');
+
+    // This lets us wait for rendering to actually occur without resolving the render promise itself.
+    // This way, we can make assertions about the initial state while also keeping the settled state
+    // suspended
+    await waitFor('[data-test-content]');
+    const element = find('[data-test-content]');
```

Comment:
For context, this is basically what you would have had to do **before** `await rerender()` was created.